src/files/infrastructure/FileJSDataverseRepository.ts

```diff
@@ -49,15 +50,17 @@ export class FileJSDataverseRepository implements FileRepository {
           jsFiles,
           FileJSDataverseRepository.getAllDownloadCount(jsFiles),
           FileJSDataverseRepository.getAllThumbnails(jsFiles),
+          FileJSDataverseRepository.getAllWithPermissions(jsFiles),
```

Comment:
If the file permissions are already obtained here, then we can remove the `getUserPermissionsById(id: number)` method from this file and from the FileRepository declaration

Also I think we can remove the checkFileDownloadPermission and checkFileEditDatasetPermission use cases